smallrye-reactive-messaging-provider/src/test/java/io/smallrye/reactive/messaging/providers/impl/ConnectorConfigTest.java

```diff
@@ -126,7 +126,7 @@ public void testGetFromEnv() {
         assertThat(config.getOptionalValue("ATTR4", String.class)).hasValue("used-2");
         // Mixed case value in env should not be found as it does not match the key we're looking for
         // either in its original casing, or after conversion to uppercase.
-        assertThat(config.getOptionalValue("mixedcase", String.class)).isEmpty();
+        assertThat(config.getOptionalValue("mixedcase", String.class)).hasValue("used");
```

Comment:
That's a change of behavior. 
I'm not sure we can accept this blindly. It's a breaking change; if we have that test, it's because it got reported. So, we are going to break at least one app.

Reply:
The assert history is coming from the original commit in https://github.com/smallrye/smallrye-reactive-messaging/pull/1017/files#diff-9826df505660d404d63487cb621c216c80b55e45ad50f25a3f93a944fc634565R26, so it doesn't seem it was added on purpose by someone that reported that.

Consider the following:

```env
FOO_bar_BAZ=1234
```

The previous implementation would match the env var with the following properties:

```properties
FOO.bar.BAZ
FOO_bar_BAZ
FOO_BAR_BAZ
```

The new implementation matches:

```properties
foo.bar.baz
FOO.bar.BAZ
FOO_bar_BAZ
FOO_BAR_BAZ
```

In my opinion, it doesn't make sense to match `FOO.bar.BAZ` and don't match `foo.bar.baz`. Effectively, they have the same semantic meaning.

Yes, I know that Env vars are case sensitive, but that something that the spec doesn't seem to care about. If that was the case, then `FOO_BAR_BAZ` shouldn't match.

Reply:
Ah got it! So, yes, it's better now.

I thought it was something that was handled and became unhandled.